### ISSUE TITLE
Cannot reproduce the zero-shot performance of LLaMA2?

### ISSUE THREAD

**Issue by A11en0** [penfever/ArcheType#5]
I use your privide code `run.py` to run the zero-shot inference on LLaMA2-13b with the prompt setting of `llama-zs` on SOTAB-27 dataset (subset). However, I just get a lower result as following:

``` python
SOTAB-27

Total entries: 220 
 Accuracy: 0.1045 
 Weighted F1: 0.142 
 Unweighted F1: 0.1371 
 Correct Remap: 16 
 Total Remap: 212
 ```
 
while the result that reported in the original paper is:

<img width="594" alt="image" src="https://github.com/penfever/ArcheType/assets/33925232/785bdaec-ca3d-4f8d-b431-a65399f902ac">

Could you help me to figure it out?

**Comment by penfever** [comment]
Without access to more information than is provided here, it's difficult to say what your issue may be -- however, it seems likely that there is, at the very least, a problem with your dataset or the path to it, as Total entries: 220 is too low for the SOTAB benchmark (it should read Total entries: 15040 when the run is complete).

**Comment by A11en0** [comment]
Thanks for your quick reply! Yep, we just use a subset of SOTAB for testing since the full inference on LLaMA2-13B is expensive for me. It could be the reason that we cannot reproduce the result. I am trying to use multiple GPU inference in this case. By the way, could you please tell me how many A100 GPUs when you conducted the zero-shot experiments for SOTAB-27 with LLaMA2-13B? and how long it take to process an instance at a time. Thanks a lot.

**Comment by A11en0** [comment]
Hi, I have try to inference on the full datasets with following setting:

bash script parameters:
``` bash
export DATA_NAME="SOTAB-27"
export MODEL_PATH="/home/yhge/pre-train/llama2-13b-hf"
export MODEL_NAME="llama-vllm-zs-ul2-frs"
export SAVE_PATH="results/llama-vllm-zs-ul2-frs-27-0116.json"

CUDA_VISIBLE_DEVICES=0,1,2,3 python src/run.py \
    --model_name=$MODEL_NAME \
    --model_path=$MODEL_PATH \
    --save_path=$SAVE_PATH \
    --input_files="$SOTAB_PATH/CTA_Test/Test" \
    --input_labels="$SOTAB_PATH/CTA_Test/CTA_test_gt.csv" \
    --label_set=$DATA_NAME \
    --method ans_contains_gt gt_contains_ans resample \
    --rules True \
    --results \
    --response \
```

generate parameters:
```python
max_length=2048,
temperature=0.5*k,
top_p=0.80-(0.1 * k),
do_sample=True,
repetition_penalty=1.3
```

and get the results as follows:

```python
Total entries: 15040
 Accuracy: 0.1844
 Weighted F1: 0.2013
 Unweighted F1: 0.1642
 Correct Remap: 2186
 Total Remap: 14425
```

It has also fallen short of achieving the results presented in the original paper. Could you please assist me with this matter?

**Comment by A11en0** [comment]
Hi, we cannot reproduce the results even we use the full dataset. How long would you open source the trained checkpoints? Thanks.

**Comment by A11en0** [comment]
By the way, we also cannot re-produce the result of Doduo on SOTAB dataset. Do you try to do this?

**Comment by penfever** [comment]
Hi, Allen, thanks for reaching out! The process for reproducing our zero-shot experimental results is described in the readme for the ArcheType repository. For example, you can run the ArcheType method zero-shot on an Alpaca-7B checkpoint using SOTAB data and SOTAB-27 labels as follows --

python3 src/run.py --model_name="alpaca-7b-zs" --save_path="results/alpaca-7b-zs-shortprompt.json" --input_files="$SOTAB_PATH/Test" --input_labels="$SOTAB_PATH/CTA_test_gt.csv" --label_set="SOTAB-27" --method ans_contains_gt gt_contains_ans resample --results --response --resume;

After the run is complete, we get the following scores.

Total entries: 15040
Accuracy: 0.1628
Weighted F1: 0.2011
Unweighted F1: 0.1891
Correct Remap: 1870
Total Remap: 14434

Here is a link to the complete experimental results we obtain after running this command.

https://drive.google.com/drive/folders/1A2PENBEVHtvOWdNSStV8gBU0W_oQ2BAf?usp=sharing

The LLAMA-2 results in Table 5 were obtained using the following command --

python src/run.py --model_name="speechless-llama2-zs-shortprompt" --save_path="/home/bf996/archetype/results/speechless-llama2-13b-zs-shortprompt.json" --input_files="$SOTAB_PATH/Test" --input_labels="$SOTAB_PATH/CTA_test_gt.csv" --label_set="SOTAB-27" --method ans_contains_gt gt_contains_ans --results --response --resume;

Unfortunately, we cannot release the weights for our fine-tuned experiment on SOTAB, as it would violate Meta's terms of service for sharing the LLAMA weights. However, we describe the process by which we attained those results in the readme of our repository.

We used the DoDuo results cited in the SOTAB paper. If you wish to reproduce them yourself, their code and paper is also available online.

________________________________

**Comment by A11en0** [comment]
Thank you for your prompt response and the detailed results provided. However, I couldn't locate the result for Alpaca-7B in the paper. It would be beneficial for me to have the zero-shot result for ArcheType-LLaMA2, as shown in Table 5, along with the original result of ArcheType fine-tuned on SOTAB-91. Sharing these results would enhance clarity and demonstrate the effectiveness of your proposed method.

**Comment by penfever** [comment]
Hi, Allen!  Thanks for checking in again, sorry for the delay in responding.


While attempting to reproduce the LLAMA-2 results, I found a bug in the contains method which was leading to worse performance using the Speechless-LLAMA2 checkpoint for zero-shot CTA. After correcting it, the following command produces the following output on our end.


python src/run.py --model_name="speechless-llama2-zs-shortprompt" --save_path="/home/bf996/archetype/results/speechless-llama2-13b-zs-shortprompt-v3.json" --input_files="$SOTAB_PATH/Test" --input_labels="$SOTAB_PATH/CTA_test_gt.csv" --label_set="SOTAB-27" --method ans_contains_gt gt_contains_ans resample --results --response --resume;

Total entries: 15040
Accuracy: 0.281
Weighted F1: 0.3
Unweighted F1: 0.2823
Correct Remap: 3639
Total Remap: 14424
The remaining small difference in F1 score between this and the paper can most likely be attributed to an efficiency change we recently implemented in the code to prevent running ArcheType on the exact same context multiple times. We do plan to update the paper with these slightly different results in a future arXiv draft.
I have added the results JSON from this run to the Google Drive link I shared with you earlier. I also added the results JSON for ArcheType fine tuning on SOTAB-91 for you to inspect at your convenience. I hope this helps address your questions!
________________________________
